RedHat_NLP/Deliverable 1.ipynb

```diff
@@ -0,0 +1,285 @@
+        "for x in range(len(array_pages)):\n",
+        "  final_array.append(base + array_pages[x] + ending)\n",
+        "\n",
+        "# pull all urls logs and store in 2-d array where array_of_logs[x] is a build-log file and \n",
+        "# array_of_logs[x][y] is an individual log line split by new line\n",
+        "array_of_logs = []\n",
+        "for x in range(len(final_array)):\n",
+        "  page = urlopen(final_array[x])\n",
+        "  html_bytes = page.read()\n",
+        "  array_of_logs.append(str(html_bytes))\n",
+        "  \n",
+        "# first log\n",
+        "print(array_of_logs[0])\n",
+        "\n",
+        "\n",
+        "# Analysis on the log data. Trying to find a framework. API (Ningxiao, Parker, Tianze, Hong)\n",
+        "#     Identify limitations with data and potential risks of achieving project goals.\n",
+        "\n",
+        "\n",
+        "# ******* PARKER *******\n",
+        "#  helper function detecting if a string is a date / timestamp\n",
+        "def is_date(str):\n",
+        "  try:\n",
+        "    dateutil.parser.parse(str)\n",
+        "    return True\n",
+        "  except:\n",
+        "    return False\n",
+        "\n",
+        "# logs variable contains all parsed logs\n",
+        "logs = []\n",
+        "for i in range(len(array_of_logs)):\n",
+        "  #  removing newline characters\n",
+        "  array_of_logs[i] = array_of_logs[i].replace('\\\\n', ' ')\n",
+        "  array_of_logs[i] = array_of_logs[i].splitlines()\n",
+        "\n",
+        "# removes leading 'b from log\n",
+        "  array_of_logs[i] = array_of_logs[i][0][2:]\n",
+        "# splitting each section as its own index (for parsing)\n",
+        "  array_of_logs[i] = array_of_logs[i].split(' ')\n",
+        "\n",
+        "#  tmp is log without timestamps\n",
+        "  tmp = []\n",
+        "  for j in range(len(array_of_logs[i])):\n",
+        "    if is_date(array_of_logs[i][j]) == False:\n",
+        "      tmp.append(array_of_logs[i][j])\n",
+        "    else:\n",
+        "      continue\n",
+        "  logs.append(tmp)\n",
+        "\n",
+        "# removes whitespace characters and keeps root words\n",
+        "stemmer = SnowballStemmer(\"english\")\n",
+        "for log in logs:\n",
+        "  log[:] = [stemmer.stem(x) for x in log if x != '']\n",
+        "\n",
+        "failures = []\n",
+        "successes = []\n",
+        "i = 0\n",
+        "for log in logs:\n",
+        "  tmp = ' '.join(log)\n",
+        "  tmp = TextBlob(tmp)\n",
+        "  sum_neg = tmp.word_counts['fail']\n",
+        "  sum_neg += tmp.word_counts['error']\n",
+        "  sum_pos = tmp.word_counts['success']\n",
+        "  sum_pos += tmp.word_counts['run']\n",
+        "  failures.append(sum_neg)\n",
+        "  successes.append(sum_pos)\n",
+        "\n",
+        "plt.xlabel('Failures')\n",
+        "plt.ylabel('Successes')\n",
+        "plt.scatter(failures, successes, s=20)\n",
+        "plt.show()\n",
```

Comment:
Once you've collected and cleaned your log messages, you should save them to your environment/repo for future use as a json or CSV file.